app.js

```diff
@@ -1,151 +0,0 @@
-document.addEventListener("DOMContentLoaded", () => {
-  let comments = [];
-  // Target DOM Elements
-  let comment_box = document.querySelector("#comment_box");
-  let comment_btn = document.querySelector("#add_comment");
-  let comment_wrapper = document.querySelector("#comment-listing");
-  let reply_textbox = document.querySelector("#reply_comment");
-  let nestedCommentForm = document.querySelector(
-    "#comment_list_inner .comment-area"
-  );
-
-  //initialize loacl stroge if not
-  if (localStorage.getItem("comments") === null)
-    localStorage.setItem("comments", JSON.stringify(comments));
-
-  //Add comment object in localStorge
-  let addComment = (comment) => {
-    let all_comments = getAllComments();
-    all_comments.push(comment);
-    updateStroge(all_comments);
-  };
-
-  //update local stroge with new data
-  let updateStroge = (newData) => {
-    localStorage.setItem("comments", JSON.stringify(newData));
-    renderComments();
-  };
-
-  //Add reply on the same index
-  let addReply = (reply, index) => {
-    let all_comments = getAllComments();
-    all_comments.forEach((element, i) => {
-      if (i == index) {
-        if (element.children) {
-          element.children.push(reply);
-        } else {
-          element.children = [];
-          element.children.push(reply);
-        }
-      }
-    });
-    updateStroge(all_comments);
-  };
-
-  //Delete parent comment
-  let deleteComment = (index) => {
-    let get_all_comments = getAllComments();
-    get_all_comments.splice(index, 1);
-    updateStroge(get_all_comments);
-  };
-
-  // Delete child comment
-  let deleteChild = (id) => {
-    let get_all_comments = getAllComments();
-    get_all_comments.map((comment) => {
```

Comment:
Not a valid usecase for `.map`. You can use `.forEach` here.

Reply:
can you explain the why forEach why not .map() ?

Reply:
Try to search for articles on when to use `.map` and when to use `.forEach`